services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -83,29 +94,52 @@
 MAX_FILES_PER_DIRECTORY = 10_000  # hf hub limitation
 MAX_OPERATIONS_PER_COMMIT = 500
 
+T = TypeVar("T")
+
 
 class ParquetFile:
-    def __init__(self, local_file: str, local_dir: str, config: str):
+    def __init__(
+        self, local_file: str, local_dir: str, config: str, split: str, shard_idx: int, partial: bool = False
+    ):
         if not local_file.startswith(local_dir):
             raise ValueError(f"{local_file} is not in {local_dir}")
         self.local_file = local_file
         self.local_dir = local_dir
         self.config = config
+        self.split = split
+        self.shard_idx = shard_idx
+        self.partial = partial
 
     @property
     def path_in_repo(self) -> str:
-        return f'{self.config}/{self.local_file.removeprefix(f"{self.local_dir}/")}'
+        if self.partial:
+            return f"{self.config}/partial/{self.split}/{self.shard_idx:04d}.parquet"
+        else:
+            return f'{self.config}/{self.local_file.removeprefix(f"{self.local_dir}/")}'
```

Comment:
you didn't try to move the other files to the same directory structure? I think it's better to manage this in another PR indeed